### PR TITLE
refactor: import gateway client and updated wallet logic

### PR DESCRIPTION
**Pull request by gregdhill**
Add example usage

### CODE REVIEW

File: sdk/examples/gateway.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        fromAddress,
+        toAddress,
+        amount,
+        fromPubKey,
+        opReturn,
+    );
+
+    // Determine how many inputs to sign
+    const inputLength = unsignedTransaction.inputsLength;
+    const _inputsToSign = Array.from({ length: inputLength }, (_, i) => i);
```
Comment (sander2) [inline sdk/examples/gateway.ts:55]: Probably should either explain why this is here, or remove this

File: sdk/examples/gateway.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    // Sign all inputs
+    const psbt = unsignedTransaction.toPSBT(0);
+    const _psbtHex = hex.encode(psbt);
+
+    // Sign all inputs with the payment address
+    const signedPsbtHex = ""; // TODO: sign PSBT
```
Comment (sander2) [inline sdk/examples/gateway.ts:62]: I'm not sure what the intended use case is here, if it's to be used in e.g. bots, or if it's to allow people to embed this in their UIs. If the latter, an example showing that would also be nice

File: sdk/src/utils.ts [hunk 3]
```diff
@@ -175,3 +180,46 @@ export function getMerkleProof(block: Block, txHash: string, forWitness?: boolea
         root: merkleAndRoot.root.toString("hex"),
     };
 }
+
+export function estimateTxFee(feeRate: number, numInputs: number = 1) {
```
Comment (sander2) [inline sdk/src/utils.ts:184]: should document the unit of the feeRate (I've think I've seen 3 different units being used in other code before)

File: sdk/src/utils.ts [hunk 4]
```diff
@@ -175,3 +180,46 @@ export function getMerkleProof(block: Block, txHash: string, forWitness?: boolea
...
+    tx.ins.map(tx_input => (tx_input.witness = [Buffer.alloc(33 + 32 + 7, 0), Buffer.alloc(33, 0)]));
+    tx.addOutput(Buffer.alloc(22, 0), 1000); // P2WPKH
+    tx.addOutput(Buffer.alloc(22, 0), 1000); // P2WPKH (change)
+    tx.addOutput(bitcoin.script.compile([bitcoin.opcodes.OP_RETURN, Buffer.alloc(20, 0)]), 0);
+    const vsize = tx.virtualSize();
+    const satoshis = feeRate * vsize;
+    return satoshis;
+}
+
+export function estimateTxFeeConst(feeRate: number, numInputs: number = 1) {
```
Comment (sander2) [inline sdk/src/utils.ts:199]: document how this is different from `estimateTxFee`?

File: sdk/src/wallet/utxo.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  toAddress: string,
+  amount: number,
+  publicKey?: string,
+  opReturnData?: string,
+  confirmationTarget: number = 3,
+): Promise<Transaction> {
+  const esploraClient = new DefaultEsploraClient(network);
+
+  // NOTE: esplora only returns the 25 most recent UTXOs
+  // TODO: change this to use the pagination API and return all UTXOs
```
Comment (sander2) [inline sdk/src/wallet/utxo.ts:45]: Do you want to address this before merging? Or maybe make an issue for it?

**Review by sander2 — COMMENTED** [review]
Assuming most of the code is copied from the current production code. Mostly lgtm but the exposed functions lack in-code function documentation

File: sdk/examples/gateway.ts (hunk 2 above)
Comment (gregdhill) [reply sdk/examples/gateway.ts:62]: This is just example code, extracted from the UI to show partners how to integrate

Comment (gregdhill) [reply sdk/examples/gateway.ts:62]: Plan is for @derrekcoleman to use this in a React example instead

File: sdk/src/utils.ts (hunk 4 above)
Comment (gregdhill) [reply sdk/src/utils.ts:199]: Removed

File: sdk/src/wallet/utxo.ts (hunk 5 above)
Comment (gregdhill) [reply sdk/src/wallet/utxo.ts:45]: Will make an issue

**Pull request merged**